File: FISTA/exp_jikkou/exp_nume_ex/.ipynb_checkpoints/exp_parameter-checkpoint.py
"""Module summary.

This is Short-Algo.py.
"""

import time
import numpy as np
import pandas as pd
import scipy.optimize as optimize
import scipy.sparse as spsp
import matplotlib.pyplot as plt
import csv
import os

from scipy.spatial import distance
from scipy.optimize import linprog
from scipy.optimize import minimize
from collections import defaultdict
from mpl_toolkits.mplot3d import Axes3D

class Parameter:
    '''モデルのパラメータを保持するクラス

    Attributes
    ----------
    K: int
        立地点数
    tau: float
        企業間交流費用パラメータ
    t: float 
        通勤費用パラメータ
    S_total: float 
        都市の総床面積
    theta_firm: float
        企業のランダム項パラメータ 
    theta_house: 
        家計のランダム項パラメータ
    T: numpy.ndarray (K, K)
        通勤費用行列
    Ker: numpy.ndarray (K, K)
        Tのカーネル行列
    D: numpy.ndarray (K, K)
        空間割引行列
    S: numpy.ndarray (K, )
        各立地点の床面積
    M: float
        総企業数
    N: float
        総家計数
    '''

    def __init__(self,
                 Col, distance_matrix, t, tau, Scaling, S_total, S_bar,
                 theta_firm, theta_house, E, RW_proj, alter_T_num,
                 alpha_1, alpha_2, beta_1, beta_2, M, N):
        '''

        Parameters
        ----------
        Num_cols: int
            立地点の列の数
        K: int
            立地点数
        distance_matrix: numpy.ndarray (K, K)
            距離行列
        tau: float
            企業間交流費用パラメータ
        t: float 
            通勤費用パラメータ
        S_total: float 
            都市の総床面積
        theta_firm: float
            企業のランダム項パラメータ 
        theta_house: float
            家計のランダム項パラメータ
        alpha_1: float
            間接効用関数のパラメータ1
        alpha_2: float
            間接効用関数のパラメータ2
        beta_1: float
            利潤関数のパラメータ1
        beta_1: float
            利潤関数のパラメータ2
        '''
        self.Col = Col
        self.K = Col * Col
        self.tau = tau
        self.t = t 
        self.Scaling = Scaling
        self.S_total = S_total
        self.S_bar = S_bar
        self.theta_firm = theta_firm
        self.theta_house = theta_house
        self.E = E
        self.RW_proj = RW_proj
        self.alpha_1 = alpha_1
        self.alpha_2 = alpha_2
        self.beta_1 = beta_1
        self.beta_2 = beta_2
        self.alter_T_num = alter_T_num
        self.M = M
        self.N = N
        self.distance_matrix = distance_matrix
        
        self.T = np.maximum(Scaling * t * alter_T_num , t * distance_matrix)
        self.D = np.exp(-tau*distance_matrix)
        
class Algo_Parameter:
    '''アルゴリズム中のパラメータを保持するクラス

    Attributes
    ----------
    L: float
            FISTAのパラメータ
    eta: float
        FISTAのパラメータ
    p_proj:
        パラメータpの射影後の値
    '''

    def __init__(self, L, eta, p_proj):
        '''

        Parameters
        ----------
        L: float
            FISTAのパラメータ
        eta: float
            FISTAのパラメータ
        p_proj:
            パラメータpの射影後の値
        '''
        self.L = L
        self.eta = eta
        self.p_proj = p_proj

File: FISTA/exp_jul/.ipynb_checkpoints/exp_parameter-checkpoint.jl
module Parameter

export Parameter, Algo_Parameter, create_parameter, create_algo_parameter

using LinearAlgebra
using SparseArrays

# モデルのパラメータを保持するクラス
struct Parameter
    Col::Int
    K::Int
    tau::Float64
    t::Float64
    Scaling::Float64
    S_total::Float64
    S_bar::Float64
    theta_firm::Float64
    theta_house::Float64
    E::Float64
    RW_proj::Float64
    alpha_1::Float64
    alpha_2::Float64
    beta_1::Float64
    beta_2::Float64
    alter_T_num::Float64
    M::Float64
    N::Float64
    distance_matrix::Matrix{Float64}
    T::Matrix{Float64}
    D::Matrix{Float64}

    function create_parameter(
        Col::Int, distance_matrix::Matrix{Float64}, t::Float64, tau::Float64, Scaling::Float64, 
        S_total::Float64, S_bar::Float64, theta_firm::Float64, theta_house::Float64, 
        E::Float64, RW_proj::Float64, alter_T_num::Float64, alpha_1::Float64, alpha_2::Float64, beta_1::Float64, 
        beta_2::Float64, M::Float64, N::Float64
    )
        K = Col * Col
        T = max.(Scaling * t * alter_T_num, t * distance_matrix)
        D = exp.(-tau * distance_matrix)

        new(
            Col, K, tau, t, Scaling, S_total, S_bar, theta_firm, theta_house, 
            E, RW_proj, alpha_1, alpha_2, beta_1, beta_2, alter_T_num, M, N, 
            distance_matrix, T, D
        )
    end
end

# アルゴリズム中のパラメータを保持するクラス
struct Algo_Parameter
    L::Float64
    eta::Float64
    p_proj::::Float64

    function create_algo_parameter(L::Float64, eta::Float64, p_proj::Float64)
        new(L, eta, p_proj)
    end
end

end # module ShortAlgo